lib/matplotlib/widgets.py

```diff
@@ -404,11 +424,31 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
         self.val = valinit
         self.valinit = valinit
         if orientation == 'vertical':
-            self.poly = ax.axhspan(valmin, valinit, 0, 1, **kwargs)
-            self.hline = ax.axhline(valinit, 0, 1, color=initcolor, lw=1)
+            self.bkd = Rectangle((.25, 0), .5, 1, transform=ax.transAxes, facecolor=bkd_color)
+            ax.add_patch(self.bkd)
+            self.poly = ax.axhspan(valmin, valinit, .25, .75, **kwargs)
+            self.hline = ax.axhline(valinit, .25, .75, color=initcolor, lw=1)
+            handleXY = [[0.5], [valinit]]
         else:
-            self.poly = ax.axvspan(valmin, valinit, 0, 1, **kwargs)
-            self.vline = ax.axvline(valinit, 0, 1, color=initcolor, lw=1)
+            self.bkd = Rectangle((0, .25), 1, .5, transform=ax.transAxes, facecolor=bkd_color)
+            ax.add_patch(self.bkd)
+            self.poly = ax.axvspan(valmin, valinit, .25, .75, **kwargs)
+            self.vline = ax.axvline(valinit, .25, .75, color=initcolor, lw=1)
+            handleXY = [[valinit], [0.5]]
+        self._handle, = ax.plot(
```

Comment:
Should the handle be a public attribute in order to enable full customizability?

Reply:
For the moment, I'd keep it private. It's a bit arcane because it's a marker represented by a one-element Line2D. This would need extra explanation. I'm also not sure if the handle should or should not scale with the slider when the figure is rescaled.

Reply:
Thats what I thought, but it is a break from the other visual elements of the slider which are public.

> I'm also not sure if the handle should or should not scale with the slider when the figure is rescaled.

I'm starting to think more and more that sliders should be specified in inches rather than whatever they are now (I guess technically in figure coords?). Though having part of the slider scale while the other doesn't seems like a bad inconsistency, so they probably ought to scale. Not sure how best to achieve marker scaling thought. (I couldn't figure out how to make a circle with consistent aspect ratio without resorting to specifying it in inches)

Reply:
> Thats what I thought, but it is a break from the other visual elements of the slider which are public.

I'd be pragmatic and wait until people ask for configurability; in particular also because we might need to change the object representing the knob if we do something about scaling (which doesn't have to be right now in this PR).

#### Size specification

> I'm starting to think more and more that sliders should be specified in inches rather than whatever they are now (I guess technically in figure coords?).

*tl;dr*: Leave as is for now.

Yes, it's figure coords. The advantage of this is that you can line up the sliders with any Axes, because they also use figure coords. IMHO inches are not a great idea, because the rendering position then depends on `dpi`. Inches is rather an abstract quantity when you render to the screen as you do with widgets (I know our figure size is in inches, but unless people are explicitly changing it, they may not even know.

#### Scaling

*tl;dr*: Leave as is for now.

- Sliders have to scale their long axis. Otherwise the layout will most certainly be broken when the figure is scaled.
- Maybe they would not necessarily have to scale their track width. But it's easier to implement scaling the track width. And it's not that bad.
- We don't have an object that stays circular under arbitrary figure scalings but also scales. Actually you have to put additional assumptions in to uniquely specify it: If the aspect ratio changes under scaling, you have two different scaling factors for x and y. How do you reduce them to one scaling factor of the circle?
One would have to write a specialized Artist for this. IMHO that's not worth it (and is anyway beyond the scope of this PR). An alternative would be using a marker and adapting it's size based on resize events (but to be checked how that interacts with draw).

  For practical reasons, stay with the circular marker. Unless people are massively scaling (> 2x) this still looks good.

Reply:
Thanks for the extensive explanations!